src/application/Application.mjs

```diff
@@ -102,12 +102,31 @@ export default class Application extends Component {
         this.__updateFocus();
     }
 
+    __focusedComponent() {
+        return this._focusPath ? this._focusPath[this._focusPath.length - 1] : undefined;
+    }
+
     __updateFocus() {
+        const prevFocusedComponent = this.__focusedComponent();
+
+        // The Application's focus path, therefore final focused component, may change here...
         const notOverridden = this.__updateFocusRec();
 
         if (!Application.booting && notOverridden) {
             this.updateFocusSettings();
         }
+
+        if (this.__options.debug) {
+            const newFocusedComponent = this.__focusedComponent();
+
+            // Testing for updateDebugFocusState safeties against lightning inspector not having been attached.
+            prevFocusedComponent && prevFocusedComponent.updateDebugFocusState && prevFocusedComponent.updateDebugFocusState();
```

Comment:
This dependency injection style of programming is out of ordinary for Lightning's codebase. Instead, you could monkey patch `__updateFocus` method to follow existing code's standards.

Another problem: inspector shouldn't be tied to `debug` flag - it should only influence the log, while inspector's `focus` attribute should operate regardless of the flag.